src/realm/sync/subscriptions.cpp

```diff
@@ -296,7 +312,31 @@ bool MutableSubscriptionSet::erase(const Query& query)
     auto ptr = find(query);
     if (!ptr)
         return false;
-    m_subs.erase(m_subs.begin() + (ptr - &m_subs.front()));
+    auto it = m_subs.begin() + (ptr - &m_subs.front());
+    erase(it);
+    return true;
+}
+
+bool MutableSubscriptionSet::erase_by_class_name(StringData object_class_name)
```

Comment:
Should we return the number of subscriptions deleted?

Reply:
Sure, if we think it's useful

Reply:
It's not necessary - the caller can always check the size before and after this call if they care about how many were removed. `bool` for success is fine